src/software/new_geom/util/intersects.cpp

```diff
@@ -1,25 +1,119 @@
 #include "software/new_geom/util/intersects.h"
 
-bool intersects(const Polygon& poly, const Segment& segment)
+#include <limits>
+
+#include "software/new_geom/util/contains.h"
+#include "software/new_geom/util/distance.h"
+#include "software/new_geom/util/intersection.h"
+
+bool intersectsNew(const Polygon &first, const Segment &second)
+{
+    for (const auto &seg : first.getSegments())
+    {
+        if (intersectsNew(seg, second))
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool intersectsNew(const Segment &first, const Polygon &second)
 {
-    for (const auto& seg : poly.getSegments())
+    return intersectsNew(second, first);
+}
+
+bool intersectsNew(const Polygon &first, const Ray &second)
+{
+    for (const auto &seg : first.getSegments())
     {
-        if (::intersects(seg, segment))
+        if (intersectsNew(seg, second))
         {
             return true;
         }
     }
     return false;
 }
 
-bool intersects(const Polygon& poly, const Ray& ray)
+bool intersectsNew(const Ray &first, const Polygon &second)
+{
+    return intersectsNew(second, first);
+}
+
+bool intersectsNew(const Polygon &first, const Circle &second)
 {
-    for (const auto& seg : poly.getSegments())
+    if (first.contains(second.getOrigin()))
     {
-        if (::intersects(seg, ray))
+        return true;
+    }
+    for (const auto &seg : first.getSegments())
+    {
+        if (distance(seg, second.getOrigin()) < second.getRadius())
         {
             return true;
         }
     }
     return false;
 }
+
+bool intersectsNew(const Circle &first, const Polygon &second)
+{
+    return intersectsNew(second, first);
+}
+
+bool intersectsNew(const Circle &first, const Circle &second)
+{
+    return (first.getOrigin() - second.getOrigin()).length() <
+           (first.getRadius() + second.getRadius());
+}
+
+bool intersectsNew(const Segment &first, const Circle &second)
+{
+    bool segment_inside_circle = containsNew(second, first);
+    double segment_start_circle_origin_distsq =
+        distanceSquared(first.getSegStart(), second.getOrigin());
+    double segment_end_circle_origin_distsq =
+        distanceSquared(first.getEnd(), second.getOrigin());
+
+    // if the segment is inside the circle AND at least one of the points is
```

Comment:
Not totally sure what this comment is referring to

Reply:
Broke up the return statement with more descriptive variable names and removed the comment since it seemed redundant. Please let me know if this is clearer!